session/session.go

```diff
@@ -215,6 +221,16 @@ func (r *Session) DoRequest(service string, method string, args []interface{}, o
 	return r.TransportHandler.DoRequest(r, service, method, args, options, pResult)
 }
 
+// AddToUserAgent allows higher level application to identify themselves by appending to the useragent string
+func (r *Session) AddToUserAgent(agent string) {
```

Comment:
I would prefer the name `AppendUserAgent`. We would also need a `ResetUserAgent` since it is the only way to go back after appending strings to the session, given the session property is package-private.

Actually, this makes me rethink doing this in the session. The reason for having it there was so that users could set it, and the default would be our choice. However, if need to hide this from the user, then there is no need to put this in the session. We can just add it as a constant in `session/rest.go` and `session/xmlrpc.go`. Unless the capability to append strings to the user agent is also needed by your team (is it?). In that case, using the session makes sense.

Reply:
Yeah the idea is to not allow the users to override the base and at the same time allow user to append to it so that it can be chained reliably. In essence, we we would like to append to the 
softlayer-go/version some value like terraform-provider-ibm/version,  for example.